community/gardening/dev-environment.md

```diff
@@ -0,0 +1,222 @@
+* Install [Minnaker](https://github.com/armory/minnaker), a POC Spinnaker instance that runs in a Linux VM on your local machine, or in the cloud.
+* [Clone and install each Spinnaker service locally.](#classic-local-installation-method)
+* [Install Spinnaker to your Kubernetes cluster](#kubernetes-installation-methods), running in your cloud provider or private cloud of choice. Read more about the [Kubernetes & Docker method](#kubernetes-and-docker-method) below.
+
+## Classic local installation method
+Follow the [Getting Set Up](https://www.spinnaker.io/guides/developer/getting-set-up/) guide to install Spinnaker locally.
+
+## Minnaker method
+If you're not sure which method to choose, we suggest following [these instructions to install Minnaker](https://github.com/armory/minnaker) in your chosen environment. This simplifies installation steps, and uses Kubernetes (K3S) under the hood. Install in a cloud VM or a local Ubuntu 18.04 VM.
+
+### Set up local development environment
+* Install your IDE. These instructions target [IntelliJ IDEA](https://www.jetbrains.com/idea/download/#section=mac).
+* Git clone the [Spinnaker service(s)](https://github.com/spinnaker) you will debug or extend.
+* Import the `gradle.properties` file from the root of the service repository into your IDE:
+  * Import Project > Select project folder > Select __Gradle__ > Click 'Finish'
+* Build the project:
+  * Open the 'Gradle' window and double-click the 'Build' task under Tasks > Builds
+
+#### __Next steps IF your Minnaker instance is running in the cloud__
+
+* [Install NGROK](https://ngrok.com/download), a tunneling service. Run it to create a tunnel from the service to the Spinnaker instance:
+  * Consult the [port mappings reference](/reference/architecture/#port-mappings) to determine which ports to forward. Create tunnels for the service(s) you're running locally.
+  * Execute `ngrok http <service port number>` e.g. `ngrok http 8089` for echo.
+  * Copy the URL in the `Forwarding` output lines.
+* Configure your Spinnaker instance to use the forwarded NGROK address(es).
+  * Create a `.hal/default/profiles/spinnaker-local.yml` file
+  * Add service settings, or copy settings from `.hal/default/staging/spinnaker.yml` and delete unnecessary services. Read more on [custom service settings](/reference/halyard/custom/#custom-service-settings).
+  * Change the `baseURL` for the service to the copied NGROK endpoint.
+* Configure the local service to communicate with the Spinnaker instance.
+  * Copy the kubeconfig from Spinnaker `/etc/spinnaker/.kube/config` to your local machine (e.g. `/tmp/kubeconfig-minnaker`)
+  * Update the kubeconfig clusters.cluster.server address to point to the external endpoint URL as in this snippet:
+
+    ```
+    apiVersion: v1
+      clusters:
+      - cluster:
+          server: ec2-34-223-57-141.us-west-2.compute.amazonaws.com:6443
+    ...
+    ```
+  * If using Minnaker, make sure the security group on your VM allows port 6443.
```

Comment:
Because we're inside the "Minnaker method" section, you can omit "If using Minnaker", right?